packages/flutter/lib/src/rendering/sliver_multi_box_adaptor.dart

```diff
@@ -78,7 +78,7 @@ abstract class RenderSliverBoxChildManager {
   /// accurate and precise non-null value. It will not be called if
   /// [createChild] is always able to create a child (e.g. for an infinite
   /// list).
-  int get childCount;
+  int? get childCount;
```

Comment:
this should be non-nullable

Reply:
In `SilverMultiBoxAdaptorElement` there's the following case that can return null:

```
int? get childCount => widget.delegate.estimatedChildCount;
```

What should we do in this case?

Reply:
I think that can be `widget.delegate.estimatedChildCount!` because childCount is only used when we're past the point where we're unable to render a child and estimatedChildCount is guaranteed to be non-null at that point.

Reply:
Just changing the return type to non-nullable and using `widget.delegate.estimatedChildCount!` causes several test failures.

Reply:
hmm, let me look into this. I'll grab your branch and investigate once I'm done with perf and relanding #63146.

Reply:
Ok, use this commit, it should solve our problems:
https://github.com/flutter/flutter/commit/00f5d1ce9897b2d353fb69df5cfccc62ebeb78db
(disclaimer: i'm still running all the tests so there's the possibility something other than the tests that failed before will now fail, but in theory the new code is just a slightly more efficient version of the old code so we should be good...)

Reply:
I cherry-picked your commit. Thanks!